### PR TITLE
Sends purchase context in payment data

### PR DESCRIPTION
**Pull request by renanporto**
#### What is the purpose of this pull request?

Adds the user accountId and the chosen deadline on payment fields for authorization. This allows the credit control payment API to capture these values in order to identify the account and issue the invoices according to the chosen deadline.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 1]
```diff
@@ -21,6 +22,39 @@ class CheckoutListener extends Component {
...
+      let deadlineIndex = this.props.chosenDeadline
+      let deadline = this.state.clientData.deadlines[deadlineIndex]
+      let purchaseContext = { accountId: accountId, deadline: deadline }
+      let encodedContext = btoa(JSON.stringify(purchaseContext))
+      this.props.onSendPayment({'card.token': encodedContext})
+    }
+  }
+
+  installmentToDeadline (installment) {
+    var count = installment.count
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: Prefer `const` or `let`

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 2]
```diff
@@ -21,6 +22,39 @@ class CheckoutListener extends Component {
...
+    var value = installment.value / 100
+    var paymentOptions = []
+    for (var i=1; i <= count; i++) {
+      paymentOptions.push({'days': 30 * i, 'value': value }) 
+    }
+    return { paymentOptions: paymentOptions }
+  }
+
+   formatDeadlines(deadlines, paymentValue) {
+      var result = deadlines.map(function(deadline) {
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: Prefer es6 notation: `deadlines.map(deadline => { `

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 3]
```diff
@@ -21,6 +22,39 @@ class CheckoutListener extends Component {
...
+      paymentOptions.push({'days': 30 * i, 'value': value }) 
+    }
+    return { paymentOptions: paymentOptions }
+  }
+
+   formatDeadlines(deadlines, paymentValue) {
+      var result = deadlines.map(function(deadline) { 
+          var paymentOptions = deadline.paymentOptions
+          var optionValue = paymentValue / paymentOptions.length 
+          var formattedOptions = paymentOptions.map(function(option) {
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: Is `formattedOptions` an array?

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 4]
```diff
@@ -29,15 +63,15 @@ class CheckoutListener extends Component {
...
     ) {
       const orderForm = window.vtexjs.checkout.orderForm
-
       const paymentValue = orderForm.value / 100
       const currencyCode = orderForm.storePreferencesData.currencyCode
       const salesChannel = parseInt(orderForm.salesChannel)
       const accountName = window.vtex.accountName
       const clientEmail = orderForm.clientProfileData.email
-
+      var ccInstallments = orderForm.paymentData.installmentOptions.find(function(i) { return i.paymentSystem === "64" })
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: `find` has two arguments, `item` & `index`. Do use `i` for `index` but not for the `item`. Easier to read.

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 5]
```diff
@@ -81,21 +115,31 @@ class CheckoutListener extends Component {
...
+          if (!creditAccounts.value || !creditAccounts.value[0]) {
             this.setState({ dataVerified: true })
             return this.manageError({
               response: { data: { code: 404 } },
               step: 'fetchDeadlines',
             })
           }
 
+          let deadlinesInfo = {}
+          if (deadlines != null && deadlines.value != null && deadlines.value.length > 0) {
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: Prefer absolute comparison `!==`

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js [hunk 6]
```diff
@@ -127,6 +171,9 @@ class CheckoutListener extends Component {
 
 CheckoutListener.propTypes = {
   render: PropTypes.func.isRequired,
+  onSendPayment: PropTypes.func.isRequired,
+  isPendingAuthorizationToFinish: PropTypes.bool,
+  chosenDeadline: PropTypes.number
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: Shouldn't it be required?

File: extension_point_creditcontrol_checkout/react/components/EnoughCredit.js [hunk 7]
```diff
@@ -5,23 +5,27 @@ import { injectIntl, intlShape } from 'react-intl'
...
       intl,
       currencyCode,
-      deadlines,
+      deadlinesInfo,
       availableCredit,
       paymentValue,
     } = this.props
-
+    let deadlines = deadlinesInfo.deadlines
+    console.log('deadlines')
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/EnoughCredit.js]: Remove logs.

File: extension_point_creditcontrol_checkout/react/components/EnoughCredit.js [hunk 8]
```diff
@@ -36,17 +40,17 @@ class EnoughCredit extends PureComponent {
                 {intl.formatMessage({ id: 'availableInstallments' })}:
               </strong>
             </p>
-
             <select onChange={this.handleChange}>
               {currencyCode &&
                 deadlines.map((installment, index) => {
                   const optionsLength = installment.paymentOptions.length
-
+                  var installmentValue = installment.paymentOptions[0].value
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/components/EnoughCredit.js]: Prefer `const` or `let`

File: extension_point_creditcontrol_checkout/react/index.js [hunk 9]
```diff
@@ -9,34 +9,64 @@ import EnoughCredit from './components/EnoughCredit'
...
+    var chosenDeadline = parseInt(info.chosenDeadline)
+    this.setState({deadline: info.chosenDeadline})
+    if (info.deadlineType === "installments") {        
+        this.props.changeInstallment({
+          installment: parseInt(chosenDeadline + 1),
+        })
+    }
   }
 
+  componentDidMount() {
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/index.js]: Empty method?

File: extension_point_creditcontrol_checkout/react/index.js [hunk 10]
```diff
@@ -9,34 +9,64 @@ import EnoughCredit from './components/EnoughCredit'
...
+
+  }
+
+  handlePayment = paymentFields => {
+      this.props.sendAuthorizationToFinish(true, [{paymentFields}])
+  }; 
+
   componentDidUpdate() {
```
Comment (kevinch) [inline extension_point_creditcontrol_checkout/react/index.js]: Empty method?

File: extension_point_creditcontrol_checkout/react/components/CheckoutListener.js (hunk 3 above)
Comment (renanporto) [reply extension_point_creditcontrol_checkout/react/components/CheckoutListener.js]: yes

**Pull request merged**